packages/runtime/container-runtime/src/summary/summaryGenerator.ts

```diff
@@ -395,14 +397,16 @@ export class SummaryGenerator {
 				cancellationToken,
 			);
 			if (waitBroadcastResult.result === "cancelled") {
-				return fail("disconnect");
+				const errorCode: SummarizeErrorCode = "disconnect";
+				return fail(errorCode, new RetriableSummaryError(getFailMessage(errorCode)));
```

Comment:
I'm not sure the value of making error required here.  Can the code that gets the retryAfter info just do `error?.retryAfterSeconds`?

Reply:
Not having error in case of failure is weird. This makes it consistent and we can update the errors to be meaningful as and when needed.